snuba/query/processors/handled_functions.py

```diff
@@ -41,7 +41,7 @@ def __init__(self, column: str, columnset: ColumnSet):
     def validate_parameters(self, exp: FunctionCall) -> None:
         validator = SignatureValidator([])
         try:
-            validator.validate(exp.parameters, self.__columnset)
+            validator.validate(exp.function_name, exp.parameters, self.__columnset)
```

Comment:
so part of the reason I didn't pass in the entity was because this `SignatureValidator` gets used in a lot of places and would need to be updated if we starting using the entity instead of the ColumnSet

Reply:
You should have easy access to the entity in every place where the SignatureValidator is used.

```
(.venv) ~/snuba  (feat/replacement_load_balancer *)$ git grep SignatureValidator
snuba/query/parser/validation/functions.py:from snuba.query.validation.signature import Any, Column, SignatureValidator
snuba/query/parser/validation/functions.py:    "like": SignatureValidator([Column({Array, String}), Any()]),
snuba/query/parser/validation/functions.py:    "notLike": SignatureValidator([Column({Array, String}), Any()]),
snuba/query/processors/custom_function.py:from snuba.query.validation.signature import ParamType, SignatureValidator
snuba/query/processors/custom_function.py:    according to the SignatureValidator.
snuba/query/processors/custom_function.py:        self.__validator = SignatureValidator(param_types)
snuba/query/processors/handled_functions.py:from snuba.query.validation.signature import SignatureValidator
snuba/query/processors/handled_functions.py:        validator = SignatureValidator([])
```

Both custom_function and handled_function are logical query processors. That means the process method has access to the entity. From the entity they can get to the columnset. 
In general in the Snuba codebase this kind of refactoring is fairly safe as it is fully typed, so mypy can tell you immediately if you messed up. I would consider making that change in its own PR that is self contained and trivial to review, which should be pretty quick.